feat: implemented pokemon details action

ChristianAsilo opened the pull request:
- Created GetPokemonDetailsAction class
- Added new list for pokemon details in AppState
- Added pokemonName in constants
![initialPokemonDetails](https://user-images.githubusercontent.com/128013689/226899963-6bf56f33-79bf-467a-9b73-495cf73f1a91.PNG)

![updated](https://user-images.githubusercontent.com/128013689/226900024-42f00064-dafa-44c4-ab23-57ce7538fc4d.PNG)

![pokemon details](https://user-images.githubusercontent.com/128013689/226883935-3fda0254-a98b-4298-9b12-178eb8f6eee4.PNG)
![analyze](https://user-images.githubusercontent.com/128013689/226883943-a4e9a8fa-77a3-48c6-b2de-7ffccf07f720.PNG)
![format](https://user-images.githubusercontent.com/128013689/226883958-acc217cc-1553-4d2e-bdc7-8ea8aab37d69.PNG)
![home](https://user-images.githubusercontent.com/128013689/226900097-f7317a33-6c40-4ad7-a5b5-eb19463d75fa.PNG)

chrstn06 committed:
feat: implemented pokemon details action
- Created GetPokemonDetailsAction class
- Added new list for pokemon details in AppState
- Added pokemonName in constants

shawn-ffufm commented on lib/home_page.dart:
```diff
@@ -9,6 +9,8 @@ class HomePage extends StatelessWidget {
   Widget build(BuildContext context) {
     StoreProvider.dispatch(context, GetPokemonsAction());
 
+    StoreProvider.dispatch(context, GetPokemonDetailsAction());
```
maybe you can add a todo comment that these are to be removed later 🙂

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -14,3 +14,13 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  @override
+  Future<AppState> reduce() async {
+    final pokemonDetails = await ApiService().pokemonApi.getPokemonDetails(name: pokemonName);
```
pokemon name should now be dynamic not fixed value, therefore you can utilize the constructor param injection here 🙂

shawn-ffufm commented on lib/state/app_state.dart:
```diff
@@ -11,6 +12,7 @@ class AppState with _$AppState {
     //TODO: this will be removed
     @Default('') String sample,
     @Default(<Pokemon>[]) List<Pokemon> pokemons,
+    @Default(<PokemonDetails>[]) List<PokemonDetails> pokemonDetails,
```
why does it need to be a list? I suggest make it a nullable `PokemonDetails` then provide a null value as its default 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,2 +1,3 @@
 const offset = '0';
 const limit = '151';
+const pokemonName = 'charizard';
```
this isn't needed anymore - you can safely remove this 🙂

shawn-ffufm reviewed (changes requested):
@ChristianAsilo see pr feedback 🙂

chrstn06 committed:
feat: implemented pokemon details action
(pr feedback)
- Added a todo comment
- Modified pokemon name into dynamic
- Modified PokemonDetails into nullable and set null value as default
- Removed pokemonName in constants

shawn-ffufm commented on lib/home_page.dart:
```diff
@@ -7,8 +7,12 @@ class HomePage extends StatelessWidget {
     StoreProvider.dispatch(context, GetPokemonsAction());
 
+    //Todo: Will be removed later
+    StoreProvider.dispatch(context, GetPokemonDetailsAction('charizard'));
```
place the charizard in a constant and have a todo comment as well 🙂

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -14,3 +14,17 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+
+///Getting of pokemon details from pokemon api
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  final String pokemonName;
```
move this below of your constructor 🙂

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -14,3 +14,17 @@ class GetPokemonsAction extends ReduxAction<AppState> {
+class GetPokemonDetailsAction extends ReduxAction<AppState> {
+  final String pokemonName;
+
+  GetPokemonDetailsAction(this.pokemonName);
```
make this a named-parameter instead 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

chrstn06 committed:
feat: implemented pokemon details action
 (pr feedback)
 - Placed charizard in a constant
 - Moved the final String pokemonName below constructor
 - Modified into named-parameter
